addons/sale_timesheet/report/project_profitability_report_analysis.py

```diff
@@ -200,23 +217,31 @@ def init(self):
                                     JOIN account_analytic_line AAL ON AAL.account_id = AA.id
-                                    LEFT JOIN account_move_line RINVL ON AAL.move_id = RINVL.id
-                                                                     AND RINVL.parent_state = 'posted'
-                                                                     AND RINVL.exclude_from_invoice_tab = 'f'
-                                    -- Check if the AAL is not related to a reversed credit note
-                                    LEFT JOIN account_move RINV ON RINV.id = RINVL.move_id
-                                    LEFT JOIN account_move_line INVL ON INVL.move_id = RINV.reversed_entry_id
-                                                                    AND INVL.parent_state = 'posted'
-                                                                    AND INVL.exclude_from_invoice_tab = 'f'
-                                                                    AND INVL.product_id = RINVL.product_id
-                                    LEFT JOIN sale_order_line_invoice_rel SOINV ON SOINV.invoice_line_id = INVL.id
+                                    LEFT JOIN account_move_line AML ON AAL.move_id = AML.id
+                                                                     AND AML.parent_state = 'posted'
+                                                                     AND AML.exclude_from_invoice_tab = 'f'
+                                    -- Check if it's not a Credit Note for an Invoice
+                                    LEFT JOIN account_move INV ON INV.id = AML.move_id
+                                    LEFT JOIN account_move_line RINVL ON RINVL.move_id = INV.reversed_entry_id
+                                                                    AND RINVL.parent_state = 'posted'
+                                                                    AND RINVL.exclude_from_invoice_tab = 'f'
+                                                                    AND RINVL.product_id = AML.product_id
+                                    -- Check if it's not a Bill reversed by a Credit Note
+                                    LEFT JOIN account_move BILL ON BILL.reversed_entry_id = AML.move_id
+                                    LEFT JOIN account_move_line BILLL ON BILLL.move_id = BILL.id
+                                                                AND BILLL.parent_state = 'posted'
+                                                                AND BILLL.exclude_from_invoice_tab = 'f'
+                                                                AND BILLL.product_id = AML.product_id
+                                    LEFT JOIN sale_order_line_invoice_rel SOINV ON SOINV.invoice_line_id = RINVL.id
                                     LEFT JOIN sale_order_line SOL ON SOINV.order_line_id = SOL.id
                                                                  AND SOL.product_id = AAL.product_id
```

Comment:
I struggle to understand why we keep those two joins if, in the `WHERE` clauses, we ensure `RINVL.id IS NULL`.

In 13.0, I think we removed the equivalent lines.